experiments/AMIP/components/atmosphere/climaatmos_extra_diags.jl

```diff
@@ -80,3 +83,95 @@ CAD.add_diagnostic_variable!(
+    standard_name = "meridional_mass_streamfunction",
+    units = "kg m^-1 s^-1",
+    compute! = (out, state, cache, time) -> begin
+        FT = eltype(state)
+        lat = CC.Fields.coordinate_field(axes(state.f)).lat
+        ρv = CC.Geometry.UVVector.(state.c.uₕ).components.data.:2 .* state.c.ρ
+        ᶠ∫_0_z_ρv = cache.scratch.ᶠtemp_scalar
+        CC.Operators.column_integral_indefinite!(ᶠ∫_0_z_ρv, ρv)
+        twoπa = FT(2π * 6371e3)
+        if isnothing(out)
+            return ᶠ∫_0_z_ρv .* twoπa .* cosd.(lat)
+        else
+            out .= ᶠ∫_0_z_ρv .* twoπa .* cosd.(lat)
+        end
+    end,
+)
+
+CAD.add_diagnostic_variable!(
+    short_name = "stab",
+    long_name = "Static Stability: N = √(g/θ dθ/dz)",
+    standard_name = "static_stability",
+    units = "s^-1",
+    compute! = (out, state, cache, time) -> begin
+        if isnothing(out)
+            return copy(static_stability(cache))
+        else
+            out .= static_stability(cache)
+        end
+    end,
+)
+
+CAD.add_diagnostic_variable!(
+    short_name = "vt",
+    long_name = "Product of meridional wind and temperature",
+    standard_name = "vt",
+    units = "K m s^-1",
+    compute! = (out, state, cache, time) -> begin
+        thermo_params = CAP.thermodynamics_params(cache.params)
+        T = TD.air_temperature.(thermo_params, cache.precomputed.ᶜts)
+        v = CC.Geometry.UVVector.(state.c.uₕ).components.data.:2
+        if isnothing(out)
+            return v .* T
+        else
+            out .= v .* T
+        end
+    end,
+)
+
+CAD.add_diagnostic_variable!(
+    short_name = "egr",
+    long_name = "(zonal) max. Eady growth rate = 0.31 f/N du/dz",
+    standard_name = "max_eady_growth_rate",
+    units = "s^-1",
+    compute! = (out, state, cache, time) -> begin
+        Ω = CAP.Omega(cache.params)
+        φ = CC.Fields.coordinate_field(state.c).lat
+        u = CC.Geometry.UVVector.(state.c.uₕ).components.data.:1
+        du_dz = cache.scratch.ᶜtemp_scalar
+        @. du_dz = CC.Geometry.WVector.(ᶜgradᵥ_.(ᶠinterp_(u))).components.data.:1
+        egr = cache.scratch.ᶜtemp_scalar_2
+        parent(egr) .= parent(FT(0.31) .* 2Ω .* sind.(φ) ./ static_stability(cache) .* du_dz)
+        if isnothing(out)
+            return copy(egr)
+        else
+            out .= egr
+        end
+    end,
+)
+
+"""
+    static_stability(cache)
+
+This function computes the static stability, or the squared root of the Brunt-Väisälä
+frequency N = √(g/θ dθ/dz) using the precomputed temperature field.
+"""
+function static_stability(cache)
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    ᶜts = cache.precomputed.ᶜts
+    θ_virt = TD.virtual_pottemp.(thermo_params, ᶜts)
+    dθ_virtdz = cache.scratch.ᶜtemp_scalar
+    @. dθ_virtdz = CC.Geometry.WVector.(ᶜgradᵥ_.(ᶠinterp_(θ_virt))).components.data.:1
+    Nsq = cache.scratch.ᶜtemp_scalar_2
+    parent(Nsq) .= parent(CAP.grav(cache.params) ./ θ_virt .* dθ_virtdz)
```

Comment:
Do we need to use `parent` here? Can we just do these calculations directly on the fields? Same for line 175 below

Reply:
You're right. As long as all gradient vectors are converted to Fields with `.components.data.:1`, we should be able to avoid parent, except the conditional 
```
        mask(N²) = N² > 0
        N = cache.scratch.ᶜtemp_scalar_3
        parent(N) .= sqrt.((parent(N²) .* mask.(parent(N²))))
```
which doesn't work with Fields.

Reply:
Why doesn't the conditional work with Fields?

Reply:
Not completely sure, but it's probably because you're relying on automatic type promotion of `mask.(N²)` from `Bool`s to floating point numbers. The `Field` broadcasting machinery is still a bit brittle when it comes to inferring type promotions certain cases. I would suggest rewriting this in a simpler way to avoid using `parent`; e.g.,
```
@. N = sqrt(max(N², 0))
```

Reply:
Great point! :)